metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableHandler.java

```diff
@@ -36,59 +36,85 @@
+    @VisibleForTesting
+    private IcebergTableCriteria icebergTableCriteria;
+    @VisibleForTesting
+    private IcebergTableOpWrapper icebergTableOpWrapper;
+    @VisibleForTesting
+    private IcebergTableRequestMetrics icebergTableRequestMetrics;
 
     /**
      * Constructor.
      *
      * @param connectorContext connector context
      */
-    public IcebergTableUtil(final ConnectorContext connectorContext) {
+    public IcebergTableHandler(final ConnectorContext connectorContext) {
         this.conf = new Configuration();
         this.connectorContext = connectorContext;
+        this.registry = connectorContext.getRegistry();
         connectorContext.getConfiguration().keySet()
             .forEach(key -> conf.set(key, connectorContext.getConfiguration().get(key)));
+        this.icebergTableCriteria = new IcebergTableCriteriaImpl(connectorContext);
+        this.icebergTableOpWrapper = new IcebergTableOpWrapper(connectorContext);
+        this.icebergTableRequestMetrics = new IcebergTableRequestMetrics(connectorContext.getRegistry());
     }
 
     /**
      * get Partition Map.
      *
-     * @param icebergTable      iceberg Table
+     * @param tableName         Qualified table name
      * @param partitionsRequest partitionsRequest
+     * @param icebergTable      iceberg Table
      * @return partition map
      */
     public Map<String, ScanSummary.PartitionMetrics> getIcebergTablePartitionMap(
-        final Table icebergTable,
-        final PartitionListRequest partitionsRequest) {
+        final QualifiedName tableName,
+        final PartitionListRequest partitionsRequest,
+        final Table icebergTable) {
+        final long start = this.registry.clock().wallTime();
+        final Map<String, ScanSummary.PartitionMetrics> result;
         try {
             if (!Strings.isNullOrEmpty(partitionsRequest.getFilter())) {
                 final IcebergFilterGenerator icebergFilterGenerator
                     = new IcebergFilterGenerator(icebergTable.schema().columns());
                 final Expression filter = (Expression) new PartitionParser(
                     new StringReader(partitionsRequest.getFilter())).filter()
                     .jjtAccept(icebergFilterGenerator, null);
-                return ScanSummary.of(icebergTable.newScan().filter(filter)).build();
+                result = this.icebergTableOpWrapper.getPartitionMetricsMap(icebergTable, filter);
+            } else {
+                result = this.icebergTableOpWrapper.getPartitionMetricsMap(icebergTable, null);
             }
         } catch (ParseException ex) {
             throw new MetacatBadRequestException("Iceberg filter parse error");
         }
-        return
-            ScanSummary.of(icebergTable.newScan())  //the top x records
-                .limit(connectorContext.getConfig().getIcebergTableSummaryFetchSize())
-                .build();
+        final long duration = registry.clock().wallTime() - start;
```

Comment:
If the above logic fails, the metrics will not be reported.

Reply:
Will move to finally block.